src/core.js

```diff
@@ -1368,7 +1368,10 @@ define([
         this.ensureCurrentMugIsSaved(function () {
             var foo = _this.getInsertTargetAndPosition(
                 _this.getCurrentlySelectedMug(), qType);
-            mug = _this.data.core.form.createQuestion(foo[0], foo[1], qType);
+            if (!foo) {
+                throw new Error("cannot add " + qType + " at the current position");
+            }
```

Comment:
There is a test that tries to insert an Item (normally a child of a Select question) at the top level and asserts that an error is thrown. Prior to this refactor somewhere in javaRosa (I think) something tried to access a property of null, which happened to throw an error in that case. However, by the time the error was thrown the mug had been created and was at least partly inserted into the vellum core data structures, leaving the form in a dirty state. This cleans up that mess.